app/src/main/java/com/internship/move/feature/authentication/LoginFragment.kt

```diff
@@ -16,11 +17,21 @@ class LoginFragment : Fragment(R.layout.fragment_login) {
         super.onViewCreated(view, savedInstanceState)
 
         binding.homeBtn.setOnClickListener {
+            updateSharedPreferences()
             findNavController().navigate(LoginFragmentDirections.actionLoginFragmentToMapGraph())
         }
 
         binding.registerBtn.setOnClickListener {
             findNavController().navigate(LoginFragmentDirections.actionLoginFragmentToRegisterFragment())
         }
     }
+
+    private fun updateSharedPreferences() {
+        val sharedPref = requireActivity().getSharedPreferences(
+            getString(R.string.preference_file_key), Context.MODE_PRIVATE) ?: return
+        with(sharedPref.edit()) {
```

Comment:
Editing shared preferences works like a builder pattern, you can chain as many calls as you want and then use `apply` or `commit` to perform those calls.
This makes the code look more concise and clean

```
requireActivity().getSharedPreferences(KEY_PREFERENCES, MODE_PRIVATE).edit()
            .putBoolean(KEY_SOME_VALUE, true)
            .putString(KEY_SOME_OTHER_VALUE, stringValue)
            .apply()
```